### PR TITLE
Stats: Take into account generated posts when determining post count.

### LINKED ISSUE #3882: Stats: Insights NUX uses a fragile test to decide if you are eligable
In #3848 we tweaked the way insights decides if we are in the nux to reflect a fix in the API to only count posts.

However, this test make niave assumptions about the number of posts a site might start with and as we introduce headstart this assumption is completely invalidated.

I think we need to change the code to use a flag set server side in the API rather than trying to infer something in the API where we have less info.

cc @kwight

### PR DESCRIPTION
**Pull request by kwight**
Sites that have been Headstarted on signup can contain up to six auto-generated posts, which shouldn't be included in the post count when deciding whether or not to display the full Insights tab.

This can't be merged until the `headstart` option has been added to the WPcom REST API `sites` endpoint.

Fixes #3882.

### CODE REVIEW

**Comment by kwight** [comment]
@sirbrillig for review

**Comment by kwight** [comment]
Committed the REST API change, `options->headstart` will show up now in API responses.

File: client/my-sites/stats/controller.js [hunk 1]
```diff
@@ -188,7 +189,12 @@ module.exports = {
 
 		analytics.pageView.record( basePath, analyticsPageTitle + ' > Insights' );
 
-		if ( site.post_count <= 1 &&
+		// Calculate the number of posts created by the user (not Headstarted) to determine if we should show the Insights tab, or the NUX placeholder.
+		postCount = site.post_count;
+		if ( site.options.headstart ) {
```
Comment (sirbrillig) [inline client/my-sites/stats/controller.js]: I don't really know how the `site` object is constructed, but to be on the safe side (like they do on line 171 above) you might want to also test for `site.options`, like `if ( site.options && site.options.headstart )`.

File: client/my-sites/stats/controller.js [hunk 2]
```diff
@@ -188,7 +189,12 @@ module.exports = {
...
+			postCount = site.post_count - site.options.headstart.original.post.length;
```
Comment (sirbrillig) [inline client/my-sites/stats/controller.js]: Similarly, on this line you access a deep value which maybe (?) could not be there. Maybe use `get` from lodash, as in `postCount = site.post_count - get( site, 'options.headstart.original.post.length', 0 )`.

Comment (sirbrillig) [inline client/my-sites/stats/controller.js]: This also assumes that `site.post_count` always exists and is an integer, but I guess we were doing that already.

Comment (kwight) [inline client/my-sites/stats/controller.js]: > This also assumes that site.post_count always exists and is an integer

Yep, this is true (as long as the user is logged in, which they need to be to be viewing Calypso).

**Comment by sirbrillig** [comment]
LGTM!

**Pull request merged**